community/values/src/main/java/org/neo4j/values/virtual/VirtualValues.java

```diff
@@ -80,26 +184,44 @@ public static EdgeReference edge( long id )
 
     public static PathValue path( NodeValue[] nodes, EdgeValue[] edges )
     {
+        assert nodes != null;
+        assert edges != null;
+        if ( nodes.length + edges.length % 2 == 0 )
```

Comment:
you want `if ( ( nodes.length + edges.length ) % 2 == 0 )` or maybe `nodes.length != edges.length + 1`?